src/Microsoft.Extensions.Http/DefaultHttpMessageHandlerBuilder.cs

```diff
@@ -9,47 +9,32 @@ namespace Microsoft.Extensions.Http
 {
     internal class DefaultHttpMessageHandlerBuilder : HttpMessageHandlerBuilder
     {
+        private readonly IHttpMessageHandlerBuilderPostInitializer _postInitializer;
+
+        public DefaultHttpMessageHandlerBuilder()
+        {
+        }
+
+        public DefaultHttpMessageHandlerBuilder(IHttpMessageHandlerBuilderPostInitializer postInitializer)
+        {
+            _postInitializer = postInitializer;
+        }
```

Comment:
The big pivot here for the first iteration is to add a 'post-initializer' which is where we hardcode logging and diagnostics. This way you don't casually break our diagnostics, you have to try if you want to do that.

If you do want to break our diagnostics, you can unregister or replace the post-initializer